### ISSUE TITLE
Material UI's withStyles and Downshift results in error "Object(...) is not a function"

### ISSUE THREAD

**Issue by dallashuggins** [downshift-js/downshift#736]
- `downshift` version: `^3.2.10`
- `node` version: `v10.15.3`
- `yarn` version: `1.13.0` 
- `material-ui` version: `^0.20.2`
- `@material-ui/core`: `^4.3.1`
- `@material-ui/styles`: `4.2.0`

**Relevant code or config**

**Form/index.js**:

*Description*: This is a part of the form component, specifically the field where we are using Downshift for the dropdown. I can provide the full component if helpful. The functions `setSubjectName` and `setSubjectId` are React Hooks. The function `getSuggestions` returns an array of objects. The `TextField` component's code is listed below this block of code. 
```javascript
<Downshift id="downshift-options">
    {({
        clearSelection,
        getInputProps,
        getItemProps,
        getLabelProps,
        getMenuProps,
        highlightedIndex,
        inputValue,
        isOpen,
        openMenu,
        selectedItem,
    })  => {
        const { onBlur, onFocus, ...inputProps } = getInputProps({
            onChange: event => {
                setSubjectName(event.target.value);
                if (event.target.value === '') clearSelection();
            },
            onSelect: event => {
                setSubjectName(event.target.value);
                const selectedSubject = arrayOfObjects.find(object => object.name === event.target.value);
                if (selectedSubject) setSubjectId(selectedSubject.subjectId);
            },
            onFocus: event => openMenu
        });
        const suggestions = getSuggestions(inputValue, {showEmpty: true});
        return (
            <>
                {TextField({
                    id: "subject-name-select",
                    value: subjectName,
                    label: (isFieldEmptyOnSubmitClick && subjectName === "") ? "You must select a subject." : "Subject Name",
                    error: (isFieldEmptyOnSubmitClick && subjectName === ""),
                    fullWidth: true,
                    type: "text",
                    InputLabelProps: getLabelProps({ shrink: true }),
                    InputProps: { onBlur, onFocus },
                    inputProps
                })}
                <div {...getMenuProps()}>
                    {isOpen ? (
                    <Paper square>
                        {suggestions.map((suggestion, index) =>
                            MenuItem({
                                value: suggestion.subjectName,
                                id: suggestion.subjectId,
                                index,
                                itemProps: getItemProps({ item: suggestion.subjectName }),
                                highlightedIndex,
                                selectedItem
                            })
                        )}
                    </Paper>
                    ) : null}
                </div>
            </>
        )
    }}
</Downshift>

```
**TextField.js**

*Description*: TextField component that is using material UI's TextField component and `withStyles` to apply style. A style object is being passed to `withStyles` - right now I'm only adding a basic object, applying just margin and color to root, just as an example. I also tried adding an object without nested properties, and have tried a lot of little things and am not seeing any results. 
```javascript
import React from 'react';
import PropTypes from 'prop-types';
import TextFieldMaterial from '@material-ui/core/TextField';
import { withStyles } from '@material-ui/styles';

const TextField = props => {
    
    const {
        component,
        classes,
        InputProps,
        inputRef = () => {},
        overrideStyles,
        ref,
        ...rest
    } = props;

    return (
        <>
            <TextFieldMaterial
                component={component}
                classes={classes}
                InputProps={{
                    inputRef: ref,
                    ...InputProps
                }}
                {...rest}
            />
        </>
    )
};

export default withStyles({
    root: {
        color: "blue",
        marginTop: 28
    }
})(TextField);
```

**What you did**:
- Added Downshift wrapper to form with a render function that renders the Material UI components TextField and MenuItem. This worked as expected without style applied. 
- Passed in a style object to `withStyles` for the `TextField` component. 
- Ran `yarn start`. 
- Open localhost and error appears in browser. 

**What happened**:

*Error*: `TypeError: Object(...) is not a function`. 

```
TypeError: Object(...) is not a function
(anonymous function)
src/components/Form/index.js:173
  170 |     }
  171 | });
  172 | const suggestions = getSuggestions(inputValue, {showEmpty: true});
> 173 | return (
      | ^  174 |     <>
  175 |         {TextField({
  176 |             id: "subject-name-select",
View compiled
▼ 17 stack frames were expanded.
Downshift.render
node_modules/downshift/dist/downshift.esm.js:1418
finishClassComponent
node_modules/react-dom/cjs/react-dom.development.js:15319
updateClassComponent
node_modules/react-dom/cjs/react-dom.development.js:15274
beginWork
node_modules/react-dom/cjs/react-dom.development.js:16262
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js:20279
workLoop
node_modules/react-dom/cjs/react-dom.development.js:20320
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js:147
invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js:196
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js:250
replayUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js:19503
renderRoot
node_modules/react-dom/cjs/react-dom.development.js:20433
performWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js:21357
performWork
node_modules/react-dom/cjs/react-dom.development.js:21267
performSyncWork
node_modules/react-dom/cjs/react-dom.development.js:21241
requestWork
node_modules/react-dom/cjs/react-dom.development.js:21096
scheduleWork
node_modules/react-dom/cjs/react-dom.development.js:20909
dispatchAction
node_modules/react-dom/cjs/react-dom.development.js:14118
```

**Problem description**:
When using the Downshift component with a render function to apply the material UI components, when `withStyles` is applied, the `Object(...) is not a function` error occurs. 

**Suggested solution**:
I tried using `makeStyles` as well and that also did not work. It resulted in an `Invalid hook call.` error that points to `const classes = useStyles(props);` which is inside the function (it's a functional component). I can provide more information but I figured one issue at a time. 

I use the `styled` functionality I can get style applied to components, but it tends to not prioritize the styles as well as the other APIs. 

Any help is greatly appreciated!

**Comment by silviuavram** [comment]
looking at a glance. have you tried supplying `itemToString` to `Downshift`? I see you are rendering object items, and `Downshift` does not know how to make a string out of them.

**Comment by dallashuggins** [comment]
Hi @silviuavram thanks for the suggestion. I passed in a `itemToString` function to the Downshift component where I console logged immediately and it never reaches that log. 

If I comment out the Downshift component from the form, it renders everything correctly, so I know it's some issue with Downshift and Material UI's `withStyles` (and same if I comment out `withStyles` and ). 

```
itemToString: item => {
    console.log("itemToString item:", item);
}
```

**Comment by silviuavram** [comment]
Did you try to add a `<div>` instead of the `<>` at the beginning? Hope this solves your problem.

The reason we need a wrapper element over that is the fact that we need to apply `getRootProps` to it. This will act as the `combobox` wrapper.

Inside it it should be the <TextField> in which you apply the `getInputProps` and the `<ul>` to which you apply `getMenuProps`. And the items each with `getItemProps`.

Also, `getLabelProps` should be applied to a text that acts as a label (not to the text input). And the label should be outside of your `combobox` div. This will act as the visible label to your combobox, and you should not need any `aria-label` whatsoever. Our getter props should take care of everything for you.

Please take a look at the [ARIA pattern](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html) and see how the DOM is structured.

Also have another look on the `Downshift` documentation how to apply all those getter props. Including `getRootProps`.

Also keep the `itemToString` prop because you are rendering objects. This prop should return a string. Probably `item.objectName`.

Let me know if I can assist you further. Good luck!

**Comment by dallashuggins** [comment]
Hi @silviuavram - Thank you so much for the level of detail in your response! I'm very sorry for the delay, I wanted to go through and try your suggestions. I did try the `<div>` in the place of a `<>` and nothing changed. I also tried working with some of the properties mentioned and didn't have any luck. I am closing this ticket because I cannot seem to get this working and there's a tight timeline for the project I'm working on, but I appreciate the help and hope to try downshift in the future.
